### PR TITLE
cleanup custom lang test matchers

### PR DESCRIPTION
**Pull request by mtoy-googly-moogly**
After some discussion @christopherswenson and I have a smaller, cleaner set of test matchers in lang that we agree on.

Been waiting for this moment to standardize, and fix all the old tests.

Now done.

### CODE REVIEW

File: packages/malloy/src/lang/test/parse-expects.ts [hunk 1]
```diff
@@ -37,11 +37,8 @@ declare global {
...
     interface Matchers<R> {
-      modelParsed(): R;
-      toBeErrorless(): R;
-      toCompile(): R;
-      modelCompiled(): R;
-      expressionCompiled(): R;
+      toParse(): R;
+      toTranslate(): R;
       toReturnType(tp: string): R;
       compileToFailWith(...expectedErrors: ErrorSpec[]): R;
```
Comment (christopherswenson) [inline packages/malloy/src/lang/test/parse-expects.ts:43]: I think this name doesn't work — maybe `translateToFailWith`?

File: packages/malloy/src/lang/test/imports.spec.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ source: botProjQSrc is from(->botProjQ)
       },
     });
     docParse.translate();
-    expect(docParse).not.toBeErrorless();
+    expect(docParse).not.toParse();
```
Comment (christopherswenson) [inline packages/malloy/src/lang/test/imports.spec.ts:96]: I think we mentioned the possibility of `toParse()` and `toTranslate()` to fail if there were warnings or errors. I *think* that `checkForErrors` in the definition of `toParse()` allows warnings — though maybe this is just something for me to fix in my PR...

Also, `.not.toParse()` would then mean "has warnings or errors." Maybe we should try not to use `.not.toParse()` and instead have a `.parseToFailWith(...)`?

**Review by christopherswenson — CHANGES_REQUESTED** [review]
Awesome. Glad we're honing in on the right answer here. Only change I feel strongly about is the name of `compileToFailWith` being something like `translateToFailWith`.

File: packages/malloy/src/lang/test/parse-expects.ts (hunk 1 above)
Comment (mtoy-googly-moogly) [reply packages/malloy/src/lang/test/parse-expects.ts:43]: I don't know how i missed this, yes `translationFailsWith` or somethiing

File: packages/malloy/src/lang/test/imports.spec.ts (hunk 2 above)
Comment (mtoy-googly-moogly) [reply packages/malloy/src/lang/test/imports.spec.ts:96]: yeah that's probably a bad test i'll fix it

**Pull request merged**